hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -650,11 +650,12 @@ public long getUsedNamespace() {
    * keys if key prefix is null.
    *
    * @param keyPrefix Bucket prefix to match
+   * @param delimiter A delimiter to group matched keys
    * @return {@code Iterator<OzoneKey>}
    */
-  public Iterator<? extends OzoneKey> listKeys(String keyPrefix)
-      throws IOException {
-    return listKeys(keyPrefix, null);
+  public Iterator<? extends OzoneKey> listKeys(String keyPrefix,
+      String delimiter) throws IOException {
```

Comment:
I think we should keep the existing signature, too, and pass `null` for `delimiter` to avoid unnecessarily breaking existing code using Ozone client API.

This would also help reduce the change by not having to add `null` in all known callers.